snowflake_connector.py

```diff
@@ -0,0 +1,445 @@
+# File: snowflake_connector.py
+#
+# Copyright (c) 2023 Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific language governing permissions
+# and limitations under the License.
+#
+
+import snowflake.connector  # isort: skip
+from snowflake_consts import *  # isort: skip
+
+import datetime
+import json
+
+# Phantom App imports
+import phantom.app as phantom
+import requests
+from phantom.action_result import ActionResult
+from phantom.base_connector import BaseConnector
+
+
+class RetVal(tuple):
+
+    def __new__(cls, val1, val2=None):
+        return tuple.__new__(RetVal, (val1, val2))
+
+
+class SnowflakeConnector(BaseConnector):
+
+    def __init__(self):
+
+        # Call the BaseConnectors init first
+        super(SnowflakeConnector, self).__init__()
+
+        self._state = None
+
+        self._account = None
+        self._username = None
+        self._password = None
+
+    def _get_error_msg_from_exception(self, e):
```

Comment:
would suggest adding a `self.error_print(tb)` in here of the full exception traceback as an error level log for ease of debugging. sometimes the exception alone doesn't give full context to what went wrong

one options is `traceback.format_exc()` which returns the tb but assumes that you're in an exception handler (which I think this function will always be since it takes an exception) 

https://docs.python.org/3/library/traceback.html#traceback.format_exc

Reply:
Great idea! I'll add it.